core/sources/domains.move

```diff
@@ -232,6 +256,16 @@ module aptos_names::domains {
 
         table::upsert(&mut aptos_names.registry, name_record_key, name_record);
 
+        let account_addr = signer::address_of(sign);
+        let reverse_lookup_result = get_reverse_lookup(account_addr);
+        if (option::is_none(&reverse_lookup_result)) {
+            // If the user has no reverse lookup set, set the user's reverse lookup.
+            set_reverse_lookup(sign, &NameRecordKeyV1 { subdomain_name, domain_name });
+        } else if (option::is_none(&subdomain_name)) {
+            // Automatically set the name to point to the sender's address
+            set_name_address_internal(subdomain_name, domain_name, signer::address_of(sign));
```

Comment:
Why do we only set the target address on the domain if the user already has a primary name? Shouldn't this always be set?

Reply:
this part is confusing - set_reverse_lookup() also sets the target address of the domain to be the user's address

Reply:
Hmm I see.